Objects/typeobject.c

```diff
@@ -4613,16 +4613,13 @@ _PyType_FromMetaclass_impl(
                 goto finally;
             }
             if (!_PyType_HasFeature(b, Py_TPFLAGS_IMMUTABLETYPE)) {
-                if (PyErr_WarnFormat(
-                    PyExc_DeprecationWarning,
-                    0,
-                    "Creating immutable type %s from mutable base %s is "
-                    "deprecated, and slated to be disallowed in Python 3.14.",
+                PyErr_Format(
+                    PyExc_TypeError,
+                    "Creating immutable type %s from mutable base %s",
```

Comment:
You can use `("Creating immutable type %s from mutable base %N", spec->name, b)` since the error message changes anyway.

See: https://peps.python.org/pep-0737/#recommend-using-the-type-fully-qualified-name